mlflow/tracing/clients/local.py

```diff
@@ -27,8 +27,41 @@ def __init__(self):
         queue_size = MLFLOW_TRACING_CLIENT_BUFFER_SIZE.get()
         self.queue = deque(maxlen=queue_size)
 
+        # used for displaying traces in IPython notebooks
+        self._prev_execution_count = -1
+
+    def _display_trace(self, trace: Trace):
```

Comment:
cc @B-Step62 updated to put the logic in the InMemoryClient instead! `_display_trace()` is wrapped entirely in a try catch so it should do nothing in a non-ipython environment